src/main/java/bio/terra/cli/serialization/command/createupdate/CreateUpdateResource.java

```diff
@@ -0,0 +1,96 @@
+package bio.terra.cli.serialization.command.createupdate;
+
+import bio.terra.workspace.model.AccessScope;
+import bio.terra.workspace.model.CloningInstructionsEnum;
+import bio.terra.workspace.model.ControlledResourceIamRole;
+import bio.terra.workspace.model.ResourceType;
+import bio.terra.workspace.model.StewardshipType;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import java.util.List;
+
+/**
+ * Parameters for creating/updating a workspace resource. This class is not currently user-facing,
+ * but could be exposed as a command input format in the future. This class handles properties that
+ * are common to all resource types. Sub-classes include additional resource-type specific
+ * properties.
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@class")
+@JsonDeserialize(builder = CreateUpdateResource.Builder.class)
+public abstract class CreateUpdateResource {
```

Comment:
optional: this abstract class doesn't have any abstract methods, so I don't think there's any behavior associated with it. I don't think we're getting any value from the is-a inheritance. Generally prefer composition over inheritance, so consider using composition instead, so that a CreateAiNotebook has-a common CreateUpdateResource data fields.

Reply:
good point about the abstract methods. changed to use a `CreateUpdateResource` property instead.